speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/unitTestUtilities/ComparerUtils.cpp

```diff
@@ -0,0 +1,560 @@
+#include "ComparerUtils.h"
+#include <gtest/gtest.h>
+#include <stdexcept>
+
+// Lifted wholesale from BlackBoxDataTests.cpp, provides comparer functions for the following structs: tCarElt, tSituation, and for tTrackSeg*
+// Does not compare values of pointers, as none of these values are being copied in BlackBoxData and are thus not useable anyhow within a black box
+
+/// @brief          Performs a test on two elements that are elementary
+/// @param a        Identifier of the first element
+/// @param b        Identifier of the second element
+/// @param p_eqOrNe A bool defined in the environment this define is called in that determines
+/// whether to compare the values of the identifiers or the pointers of the identifiers
+#define COMP_ELEM(a, b)        \
+    if (p_eqOrNe)              \
```

Comment:
'or' in the name of a boolean is not great. It is not clear what true means and what false means. (Technically `true => eq \/ ne` and `false => !eq /\ !ne` (logic ;p) Please think of a better name

Reply:
Has been renamed p_comparisonType as that is the function. Besides documentation for the functions (which already existed), I've also introduced two enum-style defines for ease of use.